compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/ToArrayLowering.kt

```diff
@@ -141,7 +139,7 @@ private class ToArrayLowering(private val context: JvmBackendContext) : ClassLow
                 toArrayName,
                 Visibilities.PUBLIC,
                 Modality.OPEN,
-                returnType = irBuiltIns.arrayClass.typeWith(irBuiltIns.anyType),
+                returnType = irBuiltIns.arrayClass.typeWith(irBuiltIns.anyNType),
```

Comment:
Could we extract nullability information from collection type?

Reply:
Sure! I'm not sure how useful it would be though, given that a user would see the frontend type `Any?` from [toArray](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-abstract-collection/to-array.html). What do you think?

Reply:
Seems not. But it's look like there could be some inconsistency if 'toArray' on call site would be binded to  lowered one with a bit wrong type. But it's not clear would it cause any problem in further ir or bytecode generation.

Reply:
Seem it's very minor for now